controllers/routingctrl/reconcile_resources.go

```diff
@@ -79,51 +79,78 @@ func (r *Controller) exportService(ctx context.Context, target *unstructured.Uns
 			return fmt.Errorf("could not load templates for type %s: %w", exportMode, err)
 		}
 
+		ownershipLabels = append(ownershipLabels, labels.ExportType(exportMode))
 		if errApply := unstruct.Apply(ctx, r.Client, resources, ownershipLabels...); errApply != nil {
 			return fmt.Errorf("could not apply routing resources for type %s: %w", exportMode, errApply)
 		}
 	}
 
-	return propagateHostsToWatchedCR(target, templateData)
+	return propagateHostsToWatchedCR(target, templateData, r.log)
 }
 
-func propagateHostsToWatchedCR(target *unstructured.Unstructured, data spi.RoutingTemplateData) error {
-	var metaOptions []metadata.Option
+func propagateHostsToWatchedCR(target *unstructured.Unstructured, data spi.RoutingTemplateData, log logr.Logger) error {
```

Comment:
Considering introductiong of `annotations.Remove` we can simplify this func too:

```golang
	exportModes := extractExportModes(target, log)

	// Remove all existing routing addresses
	metaOptions := []metadata.Option{
		annotations.Remove(annotations.RoutingAddressesExternal("")),
		annotations.Remove(annotations.RoutingAddressesPublic("")),
	}

	// TODO(mvp): put the logic of creating host names into a single place
	for _, exportMode := range exportModes {
		switch exportMode {
		case spi.ExternalRoute:
			externalAddress := annotations.RoutingAddressesExternal(fmt.Sprintf("%s-%s.%s", data.ServiceName, data.ServiceNamespace, data.Domain))
			metaOptions = append(metaOptions, externalAddress)
		case spi.PublicRoute:
			publicAddresses := annotations.RoutingAddressesPublic(fmt.Sprintf("%[1]s.%[2]s;%[1]s.%[2]s.svc;%[1]s.%[2]s.svc.cluster.local", data.PublicServiceName, data.GatewayNamespace))
			metaOptions = append(metaOptions, publicAddresses)
		}
	}

	metadata.ApplyMetaOptions(target, metaOptions...)

	return nil
```

Reply:
done, thanks for suggestion! [006ede1](https://github.com/maistra/odh-platform/pull/48/commits/006ede174dfe8cc7ef5f734525ed32153d5e9656)